Bearded.Utilities/Collections/ReversedDeletableObjectListEnumerator.cs

```diff
@@ -0,0 +1,62 @@
+using System.Collections.Generic;
+
+namespace Bearded.Utilities.Collections
+{
+    /// <summary>
+    /// Enumerator for deletable object list.
+    /// Kept internal to hide implementation.
+    /// </summary>
+    internal class ReversedDeletableObjectListEnumerator<T> : IEnumerator<T>
+        where T : class, IDeletable
+    {
+        private readonly DeletableObjectList<T> deletableObjectList;
+        private readonly List<T> list;
+        private int i;
+
+        object System.Collections.IEnumerator.Current => Current;
+        public T Current { get; private set; }
+
+        public ReversedDeletableObjectListEnumerator(DeletableObjectList<T> deletableObjectList, List<T> list)
+        {
+            this.deletableObjectList = deletableObjectList;
+            this.list = list;
+            i = list.Count;
```

Comment:
very nitpicky: should we just call `Reset()` here instead of setting `i`? This helps making sure that the resetting code between the constructor and the reset function remains in sync.